pandas/core/indexes/base.py

```diff
@@ -3953,18 +3953,37 @@ def memory_usage(self, deep: bool = False) -> int:
 
     def where(self, cond, other=None):
         """
-        Return an Index of same shape as self and whose corresponding
-        entries are from self where cond is True and otherwise are from
-        other.
+        Return a copy of self with entries replaced where cond is False.
+
+        The replacement is taken from other.
 
         Parameters
         ----------
         cond : bool array-like with the same length as self
-        other : scalar, or array-like
+            Condition to select the entries on.
+        other : scalar, or array-like, default None
+            Replacement if the condition is False.
 
         Returns
         -------
-        Index
+        pandas.Index
+            Copy of self with entries replaced from other
+            where the condition is False.
+
+        See Also
+        --------
+        DataFrame.where : Replace values in a DataFrame where the condition is False.
```

Comment:
You can say "Same method for DataFrame.". If you add `DataFrame.where` you should also add `Series.where` (better to have `Series` first, and `DataFrame` later).